### PR TITLE
[win] Elpi, Coq-Elpi and HB

### PR DESCRIPTION
**Pull request by gares**
I need to draft this since I want to test it on gitlab, it kind of works locally

TODO:
- [x] Target a released version of elpi (right now it is on a branch)
- [x] Added / updated test-suite
- [x] Last commit reverted (after green flag)
- [ ] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by gares** [comment]
Dear @Zimmi48  and @ejgallego I'd like to get this ready for the next release. May I get a deadline for this kind of PR? (I could not find much info in the wiki nor in https://github.com/coq/coq/issues/11157)

**Comment by Zimmi48** [comment]
Sorry @gares for the lack of information. I've added a tentative release schedule at https://github.com/coq/coq/wiki/Release-Plan-for-Coq-8.12, to be discussed with Emilio before becoming firm. That being said, this kind of PR has usually been accepted *after* branching / feature freeze. Ideally, it would be better to have it ready to be included in the beta, but we could even consider merging it between the beta and the final release (this has been done in the past).

**Comment by ejgallego** [comment]
Indeed, as @Zimmi48 says I see no problem in taking this kind of PR at any stage of the release process., or even in a point update.

**Comment by gares** [comment]
Screenshot from 2020-04-07 19-53-31

**Comment by gares** [comment]
FYI @CohenCyril @pi8027 I've a branch of Elpi that compiles fine under windows. Once released we can merge this and have HB bundled with Coq at every release.

**Comment by SkySkimmer** [comment]
This looks like a PR for the windows team so removing the ci-maintainers review request.

File: dev/build/windows/makecoq_mingw.sh
Comment (MSoegtropIMC) [inline dev/build/windows/makecoq_mingw.sh:1958]: It would be nice to a short description what Elpi is.

File: dev/ci/gitlab.bat
Comment (MSoegtropIMC) [inline dev/ci/gitlab.bat:61]: I think it would require some justification that these plugins are the only ones which are always build. Also if this is intended, they should got directly to the command line of MakeCoq_MinGW.bat (there used to be a few, e.g. ltac2).

File: dev/build/windows/makecoq_mingw.sh
Comment (gares) [reply dev/build/windows/makecoq_mingw.sh:1958]: wiil do

File: dev/build/windows/makecoq_mingw.sh
Comment (gares) [reply dev/build/windows/makecoq_mingw.sh:1963]: thanks, I will update this

File: dev/ci/gitlab.bat
Comment (gares) [reply dev/ci/gitlab.bat:61]: This is part of the last commit, that is meant to be removed before merging.
In this way I get CI produce the installer for me and I can download and test the artifact.

I see no reason to keep elpi and HB being built by default

Comment (gares) [reply dev/ci/gitlab.bat]: I remove this

**Comment by gares** [comment]
FTR, I could download the installer and it contains the new addons.
Using the command line utility coqtop I could load and run elpi code there.

*But* coqide seems broken, even `Check nat` makes it fail.  It is not related to this PR, I guess, but I wanted to make the release managers aware of this.

**Comment by MSoegtropIMC** [comment]
> FTR, I could download the installer and it contains the new addons.
> Using the command line utility coqtop I could load and run elpi code there.
> 
> _But_ coqide seems broken, even `Check nat` makes it fail. It is not related to this PR, I guess, but I wanted to make the release managers aware of this.

Thanks! Can you please create an issue and assign it to me?

**Comment by gares** [comment]
done. I've restarted CI, since it was stale.

**Comment by Zimmi48** [comment]
@MSoegtropIMC Seems like this PR is ready to merge?

File: dev/build/windows/makecoq_mingw.sh
Comment (MSoegtropIMC) [inline dev/build/windows/makecoq_mingw.sh:986]: Isn't it a bug in OCaml that this is not added to site-lib?

**Review by MSoegtropIMC — APPROVED** [review]
Looks good to me - will merge now.

File: dev/build/windows/makecoq_mingw.sh
Comment (gares) [reply dev/build/windows/makecoq_mingw.sh:986]: Sincerly I don't know. I think it is the opam packaging that adds this file, but I did not investigate.

**Comment by gares** [comment]
Thanks!

File: dev/build/windows/makecoq_mingw.sh
Comment (MSoegtropIMC) [reply dev/build/windows/makecoq_mingw.sh:986]: The odd thing is that I don't need this for any of the other many OCaml libs we use. So this is kind of weird and should be investigated. On the other hand we will switch to opam soon.

**Comment by Zimmi48** [comment]
Am I correct in my understanding that only Elpi and HB were tested in this PR, not the build of the rest of the add-ons?

**Comment by gares** [comment]
Yes. I tried once, and found out bignum problem IIRC. Then we set up the schedule job.

**Comment by Zimmi48** [comment]
Right! It turned out that it wasn't the only add-on which had a build issue.

**Comment by gares** [comment]
Maybe the script could be changed to not stop at the first failure?

**Comment by Zimmi48** [comment]
Indeed, that'd be nice. @MSoegtropIMC what do you think of this idea?

**Comment by MSoegtropIMC** [comment]
We are talking about the windows build script? Well for testing purposes you can remove the corresponding option at the beginning of the file but for anything release like I would prefer to have this on. One could also build in a logic to keep track of this, but since this script is anyway doomed (the near future is dune), I don't want to put effort into it.

**Pull request merged**
